src/main/java/org/apache/cassandra/sidecar/routes/CassandraHealthService.java

```diff
@@ -58,11 +60,34 @@ public CassandraHealthService(CassandraAdapterDelegate cassandra)
     @ApiResponse(responseCode = "200", description = "Cassandra is available"),
     @ApiResponse(responseCode = "503", description = "Cassandra is not available")
     })
-    @Produces(MediaType.APPLICATION_JSON)
     @GET
-    public Response getCassandraHealth()
+    @Path("/v1/cassandra/__health")
+    public Response getCassandraHealth(@Context HttpServerRequest req)
     {
-        Boolean up = cassandra.isUp();
+        final String host = req.host();
+        final CassandraAdapterDelegate cassandra = metadataFetcher.getDelegate(host.contains(":")
+                                                                               ? host.substring(0, host.indexOf(":"))
+                                                                               : host);
```

Comment:
The condition is only true for IPv4. 
In the case of IPv6, the address with port looks like this, 
```[2001:db8:0:0:0:ff00:42:8329]:9042```. Brackets are added (required), in order to distinguish the last `:`. 

Cassandra 4.0 supports IPv6. So it is a possibility to be considered.